pkg/networkservice/connectioncontext/dnscontext/client.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnscontext
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/dnscontext"
+	"github.com/networkservicemesh/sdk/pkg/tools/serialize"
+)
+
+type dnsContextClient struct {
+	cancelMonitoring context.CancelFunc
+	chainContext     context.Context
+	monitorContext   context.Context
+	coreFilePath     string
+	getCallOptions   func() []grpc.CallOption
+	dnsConfigManager dnscontext.Manager
+	monitorClient    networkservice.MonitorConnectionClient
+	executor         serialize.Executor
+}
+
+// NewClient creates a new DNS client chain component. Setups all DNS traffic to the localhost. Monitors DNS configs from connections.
+func NewClient(chainContext context.Context, coreFilePath, resolveConfigPath string, monitorClient networkservice.MonitorConnectionClient, getMonitorCallOptions func() []grpc.CallOption) networkservice.NetworkServiceClient {
+	c := &dnsContextClient{
+		coreFilePath:     coreFilePath,
+		chainContext:     chainContext,
+		dnsConfigManager: dnscontext.NewManager(),
+		monitorClient:    monitorClient,
+		getCallOptions:   getMonitorCallOptions,
+	}
+	if r, err := dnscontext.OpenResolveConfig(resolveConfigPath); err != nil {
+		logrus.Errorf("DnsContextClient: can not load resolve config file. Path: %v. Error: %v", resolveConfigPath, err.Error())
+	} else {
+		c.dnsConfigManager.Store("", &networkservice.DNSConfig{
+			SearchDomains: r.Value(dnscontext.AnyDomain),
+			DnsServerIps:  r.Value(dnscontext.NameserverProperty),
+		})
+		r.SetValue(dnscontext.NameserverProperty, "127.0.0.1")
```

Comment:
Should we support IPv6 here as well, or it is not required?

Reply:
Thanks, fixed.